segtypes/code.py

```diff
@@ -397,7 +397,7 @@ def split(self, rom_bytes, base_path):
                             out_lines.extend(funcs_text[func][0])
                             out_lines.append("")
 
-                            outpath = Path(os.path.join(out_dir, split_file["name"], func_name + ".s"))
+                            outpath = Path(os.path.join(out_dir, split_file["name"], func_name + "." + self.get_exit(split_file["subtype"])))
```

Comment:
The calls to `get_exit` should probably be `get_ext`, and this one in particular shouldn't happen since we explicitly want to write .s files despite `split_file["subtype"]c` being "c"